### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by nuclearsandwich** [open-rmf/rmf_visualization#60]
### Before proceeding, is there an existing issue or discussion for this?

- [X] I have done a search for similar [issues](.?q=type%3Aissue) and [discussions](https://github.com/open-rmf/rmf/discussions).

### OS and version

Ubuntu 22.04

### Open-RMF installation type

Binaries

### Open-RMF version or commit hash

2.2.0-1

### ROS distribution

Rolling

### ROS installation type

Binaries

### Description of the bug

The following compilation unit begins building at 4minutes into the build and is still running when the build hits the general timeout for build farm packages at 120 minutes.

```
00:04:23.552  /usr/lib/ccache/c++ -fPIC -g -O2 -ffile-prefix-map=/tmp/binarydeb/ros-rolling-rmf-visualization-schedule-2.2.0=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -shared -Wl,-soname,libschedule_visualizer.so -o libschedule_visualizer.so CMakeFiles/schedule_visualizer.dir/src/ScheduleVisualizer.cpp.o  -Wl,-rpath,/tmp/binarydeb/ros-rolling-rmf-visualization-schedule-2.2.0/.obj-x86_64-linux-gnu:/opt/ros/rolling/lib:/opt/ros/rolling/lib/x86_64-linux-gnu: librmf_visualization_schedule.so /opt/ros/rolling/lib/libcomponent_manager.so /opt/ros/rolling/lib/libament_index_cpp.so /usr/lib/x86_64-linux-gnu/libconsole_bridge.so.1.0 /opt/ros/rolling/lib/librcpputils.so /opt/ros/rolling/lib/libclass_loader.so /usr/lib/x86_64-linux-gnu/libconsole_bridge.so.1.0 /lib/libstd_msgs__rosidl_typesupport_fastrtps_c.so /opt/ros/rolling
    trimmed due to GitHub comment length limits. This was 124k of CLI args
02:00:00.101  Build timed out (after 120 minutes). Marking the build as failed.
```

### Steps to reproduce the bug

The latest failed build is https://build.ros2.org/view/Rbin_uJ64/job/Rbin_uJ64__rmf_visualization_schedule__ubuntu_jammy_amd64__binary/107/

This started on July 11.

### Expected behavior

 The most [recent successful build](https://build.ros2.org/view/Rbin_uJ64/job/Rbin_uJ64__rmf_visualization_schedule__ubuntu_jammy_amd64__binary/103/) on July 6 is also version 2.2.0-1 and completed in 81 minutes.

The shared object `libschedule_visualizer.so` appears to have a build time of 1h 16m during a normal build which has now grown even further, perhaps due to changes in dependencies.

```
00:03:49.117  /usr/lib/ccache/c++ -fPIC -g -O2 -ffile-prefix-map=/tmp/binarydeb/ros-rolling-rmf-visualization-schedule-2.2.0=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -shared -Wl,-soname,libschedule_visualizer.so -o libschedule_visualizer.so CMakeFiles/schedule_visualizer.dir/src/ScheduleVisualizer.cpp.o  -Wl,-rpath,/tmp
        Trimmed due to GitHub comment length limits
01:19:40.061  make[4]: Leaving directory '/tmp/binarydeb/ros-rolling-rmf-visualization-schedule-2.2.0/.obj-x86_64-linux-gnu'
```

### Additional information or screenshots

The first time I tried submitting this bug the log entries exceeded the comment length and broke GitHub so bad I had to refresh.

**Comment by Yadunund** [comment]
Thanks for reporting this! In the past I've noticed that build times are dominated by linking and we found a significant decrease in build time after switching to the `lld` linker. Hence, we even explicitly ask users who [build from source](https://github.com/open-rmf/rmf#building-from-source) to switch to `lld` from the default `ld` linker. I'm assuming this not something we can change on the buildfarm side?
But it's also worth looking at our codebase to understand why linking times have increased.
